dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -168,6 +183,381 @@ class ComputeServerRunner : public DB::tests::MPPTaskTestUtils
         auto properties = DB::tests::getDAGPropertiesForTest(serverNum(), 1, 1, query_ts);
         addOneGather(running_queries, gather_ids, properties);
     }
+
+    // To test auto pass through hashagg.
+    struct AutoPassThroughTestData
```

Comment:
Does this case cover all possible state switches for auto pass through feature, like  passThrough => adjust, selective => adjust and so on?

Reply:
Yes. check `ComputeServerRunner.stateSwitchMediumNDV`